src/highlight/HighlightList.tsx

```diff
@@ -1,20 +1,62 @@
 import React from 'react';
 import classNames from 'classnames';
-import HighlightAnnotation from './HighlightAnnotation';
+import HighlightCanvas, { CanvasShapeRect } from './HighlightCanvas';
+import HighlightSvg from './HighlightSvg';
+import HighlightTarget from './HighlightTarget';
 import useOutsideEvent from '../common/useOutsideEvent';
-import { AnnotationHighlight } from '../@types';
-
+import { AnnotationHighlight, Rect, TargetHighlight } from '../@types';
+import { checkValue } from '../utils/util';
 import './HighlightList.scss';
 
 export type Props = {
+    activeId?: string | null;
     annotations: AnnotationHighlight[];
     className?: string;
     onSelect?: (annotationId: string | null) => void;
 };
 
-export function HighlightList({ annotations, className, onSelect }: Props): JSX.Element {
+export function filterHighlight({ target }: { target: TargetHighlight }): boolean {
+    const { shapes = [] } = target;
+
+    return shapes.reduce((isValid: boolean, rect: Rect) => {
+        const { height, width, x, y } = rect;
+        return isValid && checkValue(height) && checkValue(width) && checkValue(x) && checkValue(y);
+    }, true);
+}
+
+export function getHighlightArea(shapes: Rect[]): number {
+    return shapes.reduce((area, { height, width }) => area + height * width, 0);
+}
+
+export function sortHighlight(
+    { target: targetA }: { target: TargetHighlight },
+    { target: targetB }: { target: TargetHighlight },
+): number {
+    const { shapes: shapesA } = targetA;
+    const { shapes: shapesB } = targetB;
+
+    // Render the smallest highlights last to ensure they are always clickable
+    return getHighlightArea(shapesA) > getHighlightArea(shapesB) ? -1 : 1;
+}
+
+export function getRectsFromAnnotations(annotations: AnnotationHighlight[]): CanvasShapeRect[] {
```

Comment:
We're using "rect" and "shape" kind of interchangeably and in combination. We should try to keep the naming and types consistent within components when possible. It seems like these should be Shapes?